app/battle/battlefront.py

```diff
@@ -0,0 +1,31 @@
+from app.battle.knight import Knight
+
+
+def battle(knights_config: dict) -> dict:
+    lancelot = Knight(**knights_config["lancelot"])
+    arthur = Knight(**knights_config["arthur"])
+    mordred = Knight(**knights_config["mordred"])
+    red_knight = Knight(**knights_config["red_knight"])
+
+    battle1 = fight(lancelot, mordred)
+    battle2 = fight(arthur, red_knight)
+
+    return {
+        **battle1,
+        **battle2,
```

Comment:
Merging dictionaries using `**` could lead to key conflicts if two knights have the same name. Consider using a more explicit method to handle the results, such as appending results to a list or ensuring unique keys.